portingdb/htmlreport.py

```diff
@@ -588,6 +589,82 @@ def ascending(p):
     )
 
 
+def mispackaged():
+    # Parameters
+    requested = request.args.get('requested', None)
+    if requested not in ('1', None):
+        abort(400)  # Bad request
+
+    # Database
+    db = current_app.config['DB']()
+    query = queries.packages(db)
+
+    # Get data
+    mispackaged, query = queries.split(query, tables.Package.status == 'mispackaged')
+    mispackaged = mispackaged.options(subqueryload('collection_packages'))
+    mispackaged = mispackaged.options(subqueryload('collection_packages.links'))
+    mispackaged = mispackaged.options(subqueryload('collection_packages.tracking_bugs'))
+
+    if requested:
+        # Filter only to packages where maintainer requested a patch
+        mispackaged = [pkg for pkg in mispackaged
+                if "https://bugzilla.redhat.com/show_bug.cgi?id=1333765"
+                in pkg.list_tracking_bugs]
+
+    # Sorting the packages based on their last_link_update
+    #   Note that sorting through the database is problematic, because
+    #   to sort on a @property you need to create a @hybrid_property which
+    #   needs to be written as sort of an SQL expression
+
+    # Packages without a BZ report will be last
+    nonevalue = datetime.datetime.utcnow()
+    mispackaged = sorted(mispackaged,
+            key=lambda pkg: pkg.last_link_update or nonevalue)
+
+    # Render the page, pass the data
+    return render_template(
+        'mispackaged.html',
+        breadcrumbs=(
+            (url_for('hello'), 'Python 3 Porting Database'),
+            (url_for('mispackaged'), 'Mispackaged'),
+        ),
+        requested=bool(requested),
+        mispackaged=mispackaged,
+        ago=ago,
+    )
+
+def ago(date):
+    """Displays roughly how long ago the date was in a human readable format"""
+    now = datetime.datetime.utcnow()
+    diff = now - date
+
+    # Years
+    if diff.days >= 365:
+        if diff.days >= 2 * 365:
+            return "{} years ago".format(math.floor(diff.days / 365))
+        else:
+            return "a year ago"
```

Comment:
Technically yeah. I didn't work in rounding for the years as you really don't need a big precision there. The point of the list is ordering by time, the specific values are just for quick orientation. I'm open to PRs if you think it's important :)